### PR TITLE
modifying categorization tests to warn instead of error, improved rep…

### PR DESCRIPTION
**Pull request by mhines-usgs**
…ort appearance and content

### CODE REVIEW

**Comment by mhines-usgs** [comment]
from an email chain in April, we discussed allowing the pipeline to continue even if a newly Undefined HRU appears, since our tile generation pipeline ignores the category Undefined, we won't see those on the final tilesets anyway.

the rest of the changes are cosmetic/appearance-related as well as beefing up the utility of the report by including additional icons and data. a new example report output can be found here. https://wbeep-test-website.s3-us-west-2.amazonaws.com/estimated-availability/date/test_results/order_of_magnitude_test_2021-04-09.html

**Comment by lindsayplatt** [comment]
@mhines-usgs do you have logs to show the messages when `it actually still is reporting a problem`? Just curious what it is returning

**Comment by mhines-usgs** [comment]
I just ran this locally and uploaded the output toS3. The data from 3/31/2021 didn't have any new Undefined values, but the 4/9/2021 date did. You can see the different output for those validations (that don't error and stop the pipeline now) in the reports here at the very bottom. my mistake previously was confusing the TRUE/FALSE messaging in the validate_that function.

https://wbeep-test-website.s3-us-west-2.amazonaws.com/estimated-availability/date/test_results/order_of_magnitude_test_2021-04-09.html

https://wbeep-test-website.s3.us-west-2.amazonaws.com/estimated-availability/date/test_results/order_of_magnitude_test_2021-03-31.html

File: src/validate_total_storage_categorized.R [hunk 1]
```diff
@@ -8,34 +8,38 @@ validate_total_storage_categorized <- function(data_categorized, validate_fn, n_
...
+  if (assert_that(is.character(data_categorized$value))) {
+    write(x = sprintf("<img src='icon-check.png' alt='test passed'> Category data are in the expected character format.<br />"), 
+          file = validate_fn,
+          append = TRUE)
+  }
   # Use validate_that instead of assert_that so we get a warning instead of an error
-  validate_that(all(data_categorized_problem_hru$value == "Undefined"), msg = "All the HRUs categorized as problems are correctly categorized as Undefined.")
-  validate_that(!"Undefined" %in% unique(data_categorized_good_hrus$value), msg = "There are unexpected/new HRUs with the categorization of Undefined. ")
+  validate_that(all(data_categorized_problem_hru$value == "Undefined"), msg = "Expecting all Undefined categorization, other value found.")
+  validate_that(!"Undefined" %in% unique(data_categorized_good_hrus$value), msg = "Expecting all properly categorized data, Undefined found")
```
Comment (lindsayplatt) [inline src/validate_total_storage_categorized.R:37]: Where do these two messages end up? Just as console lines and not part of the report?

File: src/validate_total_storage_categorized.R [hunk 2]
```diff
@@ -8,34 +8,38 @@ validate_total_storage_categorized <- function(data_categorized, validate_fn, n_
...
   all_new_problem_hrus <- values_categorized[which(values_categorized$value=='Undefined'),]
   all_new_problem_hrus <- all_new_problem_hrus$hru_id_nat
```
Comment (lindsayplatt) [inline src/validate_total_storage_categorized.R:39]: for clarity's sake, seems like this should be named `all_problem_hrus` so as not to confuse with the "new" ones.

Comment (lindsayplatt) [reply src/validate_total_storage_categorized.R:39]: or better yet `all_undefined_hrus`

**Comment by lindsayplatt** [comment]
Glad you figured out the issue with needing to add `if()` statements around the `assert_that()` and `write()` calls. It all looks good to me based on the code functioning. Just two smaller comments

File: src/validate_total_storage_categorized.R (hunk 1 above)
Comment (mhines-usgs) [reply src/validate_total_storage_categorized.R:37]: hmm yes, I'll get them to output too, somehow I missed them :)

**Review by lindsayplatt — APPROVED** [review]
Makes sense now!

**Pull request merged**